helios-services/src/main/java/com/spotify/helios/agent/TaskRunner.java

```diff
@@ -115,6 +115,13 @@ public void stop() throws InterruptedException {
     // Interrupt the thread blocking on waitContainer
     stopAsync().awaitTerminated();
 
+    if (System.getenv("CONTAINER_STATS") != null) {
+      try {
+        log.info("container {} stats: {}", containerName, docker.stats(containerName));
+      } catch (DockerException e) {
+        log.warn("Could not log container stats. Exception was {}", e);
```

Comment:
Not critical, but you should leave out the `{}` when passing an exception to slf4j. It will end up logging something like:

```
WARN Could not log container stats. Exception was {}
java.lang.WhateverTheExceptionWas:foobar
  at ...
  at ...
```

because there is an overload for each Logger method that takes a Throwable as the last parameter: https://www.slf4j.org/api/org/slf4j/Logger.html#warn(java.lang.String,%20java.lang.Throwable)

also probably more idiomatic to just have the message look like

```java
log.warn("Could not log container stats", ex);
```

Reply:
Ah, thanks for the tip.